rdflib/plugins/stores/sparqlconnector.py

```diff
@@ -94,47 +92,44 @@ def query(self, query, default_graph=None):
 
         if self.method == "GET":
             args["params"].update(params)
+            qsa = "?" + urlencode(args["params"])
+            res = urlopen(Request(self.query_endpoint + qsa, headers=args["headers"]))
         elif self.method == "POST":
             args["headers"].update({"Content-Type": "application/sparql-query"})
-            args["data"] = params
+            try:
+                res = urlopen(Request(self.query_endpoint, data=query.encode(), headers=args["headers"]))
+            except HTTPError as e:
+                return e.code, str(e), None
         else:
             raise SPARQLConnectorException("Unknown method %s" % self.method)
 
-        res = self.session.request(self.method, **args)
-
-        res.raise_for_status()
-
         return Result.parse(
-            BytesIO(res.content), content_type=res.headers["Content-type"]
+            BytesIO(res.read()), content_type=res.headers["Content-Type"].split(";")[0]
         )
 
-    def update(self, update, default_graph=None):
+    def update(self, query, default_graph: str = None, named_graph: str = None):
         if not self.update_endpoint:
             raise SPARQLConnectorException("Query endpoint not set!")
 
         params = {}
 
-        if default_graph:
+        if default_graph is not None:
             params["using-graph-uri"] = default_graph
 
+        if named_graph is not None:
+            params["using-named-graph-uri"] = default_graph
+
         headers = {
             "Accept": _response_mime_types[self.returnFormat],
             "Content-Type": "application/sparql-update",
         }
 
-        args = dict(self.kwargs)
-
-        args.update(url=self.update_endpoint, data=update.encode("utf-8"))
+        args = dict(self.kwargs)  # other QSAs
 
-        # merge params/headers dicts
         args.setdefault("params", {})
         args["params"].update(params)
         args.setdefault("headers", {})
         args["headers"].update(headers)
 
-        res = self.session.post(**args)
-
-        res.raise_for_status()
-
-    def close(self):
```

Comment:
Why do we remove `close()` method?

Reply:
Because the Store is actually stateless with each operation and there's nothing to close. I originally dropped `open()` too but it's needed if the Store is called by `Graph("SPARQLStore")` as opposed to `SPARQLStore()`